scripts/deploy.py

```diff
@@ -0,0 +1,497 @@
+    p.AddStep(CompareFileSha(OWN_FILENAME, require_equal=True))
+
+
+def UpdateChangeLog(p):
+    p.AddStep(SaveGitLog)
+    p.AddStep(PrependGitLogToChangeLog)
+    p.AddStep(RunStoreFileSha(CHANGELOG_PATH))
+    p.AddStep(
+        RunCmdStep(
+            '%s %s' % (os.getenv('EDITOR', 'vim'), CHANGELOG_PATH),
+            doc="Edit changelog"))
+    p.AddStep(CompareFileSha(CHANGELOG_PATH, require_equal=False))
+    p.AddStep(RunCmdStep('git add %s' % CHANGELOG_PATH))
+
+
+############################################################################
+# Pipeline machinery
+############################################################################
+
+
+def RunJump(where):
+    def f(ctx):
+        raise Jump(where)
+
+    if where > 0:
+        f.__doc__ = "Go %d steps forward" % where
+    else:
+        f.__doc__ = "Go %d steps backwards" % -where
+    return f
+
+
+def FormatVersion(version_dict):
+    res = '%d.%d.%d' % (
+        version_dict['MAJOR'],
+        version_dict['MINOR'],
+        version_dict['PATCH'],
+    )
+    if version_dict['POSTFIX']:
+        res += '-%s' % version_dict['POSTFIX']
+    return res
+
+
+def SelectionPrompt(caption, choices, *, zero_choice=None):
+    while True:
+        click.secho(caption)
+        if zero_choice:
+            click.secho("%3d. %s" % (0, zero_choice), fg='yellow')
+        for i, n in enumerate(choices):
+            click.secho("%3d. %s" % (i + 1, n[0]), fg='yellow')
+        r = click.prompt('', prompt_suffix='>>>>>>> ')
+        try:
+            r = int(r) - 1
+            if zero_choice and r == -1:
+                return None
+            if 0 <= r < len(choices):
+                return choices[r][1]
+        except:
+            pass
+
+
+def SelectCommitToCherrypick(jump_on_end):
+    def f(ctx):
+        proc = subprocess.Popen(
+            ['git', 'log', 'HEAD..master', '--pretty=format:%h %s'],
+            stdout=subprocess.PIPE)
+        commits = proc.communicate()[0].decode('utf-8').split('\n')
+        if proc.returncode != 0:
+            click.secho('Git returned non-zero exit code!', fg='red')
+            return False
+        ctx['cherrypick-commit'] = SelectionPrompt(
+            "Select commit to cherrypick.",
+            [(x, x.split()[0]) for x in commits],
+            zero_choice="[Select 0 to finish.]")
+        if ctx['cherrypick-commit'] is None:
+            raise Jump(jump_on_end)
+        return True
+
+    f.__doc__ = "Choose which commit to cherrypick"
+    return f
+
+
+def CherryPick(ctx):
+    """Cherry-pick a commit"""
+    return RunCmdStep('git cherry-pick %s' % ctx['cherrypick-commit'])(ctx)
+
+
+def WriteVersionToFile(version):
+    with open(VERSION_PATH, "w") as f:
+        f.write('// This file is automatically generated by deploy.py. '
+                'Do not edit this file.\n')
+        for k, v in version.items():
+            f.write('#define LC0_VERSION_%s %s\n' % (k, json.dumps(v)))
```

Comment:
Added comment.
But in general it's because it will handle both strings correctly (including ones with special character etc, like `"asd\"sdf\n"`) and numbers.
`repr()` will use single quotes instead.